ONSdigital/florence
# Feature/search api for release calendar entry

### What

Previously florence was using babbage to get release calendar entries, but due to decommissioning the release calendar in babbage this needed to be switched to use the search api [instead](https://jira.ons.gov.uk/browse/DIS-1106). 

### How to review

- Grab this branch
- Run the search stack
- Run florence
- Login
- Check "Calendar entry schedule"
- Click "Select a calendar entry"
and if all is working, you should see release calendar entries

![image](https://github.com/user-attachments/assets/5c44541e-42dd-47fc-b226-2b8a6480183f)
![image](https://github.com/user-attachments/assets/fccd3933-6621-413c-be85-2a7685850d62)

You can also check it against sandbox. By changing `APIRouterURL` in config to `https://api.dp.aws.onsdigital.uk` then doing `make debug` again while still logged in and refreshing. The calendar entries should now pull from sandbox.

![image](https://github.com/user-attachments/assets/fdfd9d60-8df3-4a56-b123-688addf76a00)

* these entries won't directly match current sandbox florence, as currently babbage pulls all releases that are nott published and no cancelled, while the search api returns the not published, not cancelled _AND_ from date/time now.

### Who can review

Florence dev
